sssom/io.py

```diff
@@ -1,78 +1,70 @@
-    """
+def convert_file(
+    input_path: str,
+    output_path: Optional[str] = None,
+    output_format: Optional[str] = None,
+) -> None:
+    """Convert a file.
 
     Args:
         input_path: The path to the input SSSOM tsv file
-        output_path: The path to the output file.
+        output_path: The path to the output file. If none is given, will default to using stdout.
         output_format: The format to which the the SSSOM TSV should be converted.
-
-    Returns:
-
     """
-    if validators.url(input_path) or os.path.exists(input_path):
-        doc = read_sssom_table(input_path)
-        write_func, fileformat = get_writer_function(output_format, output_path)
-        write_func(doc, output_path, serialisation=fileformat)
-    else:
-        raise Exception(f"{input_path} is not a valid file path or url.")
+    raise_for_bad_path(input_path)
+    doc = read_sssom_table(input_path)
+    write_func, fileformat = get_writer_function(
+        output_format=output_format, output=output_path
+    )
+    write_func(doc, output_path, serialisation=fileformat)
 
 
 def parse_file(
     input_path: str,
-    output_path: str = None,
-    input_format: str = None,
-    metadata_path: str = None,
-    curie_map_mode: str = None,
+    output_path: Optional[str] = None,
+    input_format: Optional[str] = None,
+    metadata_path: Optional[str] = None,
+    curie_map_mode: Optional[str] = None,
     clean_prefixes: bool = True,
-):
+) -> None:
     """
 
     Args:
-        input_path (str): The path to the input file in one of the legal formats, eg obographs, aligmentapi-xml
-        output_path (str): The path to the output file.
-        input_format (str): The string denoting the input format.
-        metadata_path (str): The path to a file containing the sssom metadata (including curie_map)
-         to be used during parse.
-        curie_map_mode (str): Defines wether the curie map in the metadata should be extended or replaced with
-         the SSSOM default curie map. Must be one of metadata_only, sssom_default_only, merged
-        clean_prefixes (bool): If True (default), records with unknown prefixes are removed from the SSSOM file.
-    Returns:
-
+        input_path: The path to the input file in one of the legal formats, eg obographs, aligmentapi-xml
+        output_path: The path to the output file.
+        input_format: The string denoting the input format.
+        metadata_path: The path to a file containing the sssom metadata (including curie_map)
+            to be used during parse.
+        curie_map_mode: Defines whether the curie map in the metadata should be extended or replaced with
+            the SSSOM default curie map. Must be one of metadata_only, sssom_default_only, merged
+        clean_prefixes: If True (default), records with unknown prefixes are removed from the SSSOM file.
     """
-    if validators.url(input_path) or os.path.exists(input_path):
```

Comment:
any time you have

```python
if cond:
    ...  # big code block
else:
    raise ...
```

Switch to `if not cond`, then you can dedent the entire block. If you turn your head 90 degrees and your code has many mountainous peaks and valleys, there's a problem

Reply:
Hard to remember this great advice, can this be checked by the code readbility checkers?

Reply:
It could be added as a flake8 plugin in theory, but I don’t know enough on how to make one myself